Compile failed on Mac

**What version of Rust are you using?**
rustc 1.32.0-nightly (14997d56a 2018-12-05)

**What operating system and CPU are you using?**
macOS 10.14.2 (18C54)
Intel Core i7

**What did you do?**
Just run ```make build```.

**What did you expect to see?**
Build successfully.

**What did you see instead?**
```
error[E0308]: mismatched types
   --> /Users/sixedge/.cargo/git/checkouts/procinfo-rs-d811f15b3a9969c8/ac3841d/src/pid/status.rs:195:1
    |
195 | named!(parse_umask<mode_t>,     delimited!(tag!("Umask:\t"),     parse_u32_octal,    line_ending));
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected u16, found u32
    |
    = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: Could not compile `procinfo`.
```

I think this submission ```* c8ff5697 util/metrics: use procinfo-rs (#3820)``` brought this error.

sudo make build
Password:
cargo build --features "default sse no-fail"
   Compiling procinfo v0.4.2 (https://github.com/tikv/procinfo-rs#ac3841d2)
   Compiling librocksdb_sys v0.1.0 (https://github.com/pingcap/rust-rocksdb.git#1317e2ed)
error[E0425]: cannot find function `parse_umask` in this scope
   --> /Users/momo/.cargo/git/checkouts/procinfo-rs-d811f15b3a9969c8/ac3841d/src/pid/status.rs:272:18
    |
272 |                | parse_umask        => { |value| status.umask       = value }
    |                  ^^^^^^^^^^^ did you mean `parse_u32s`?

   Compiling rocksdb v0.3.0 (https://github.com/pingcap/rust-rocksdb.git#1317e2ed)
error: aborting due to previous error

For more information about this error, try `rustc --explain E0425`.
error: Could not compile `procinfo`.

To learn more, run the command again with --verbose.
make: *** [build] Error 101

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -106,7 +106,6 @@ rust-crypto = "^0.2"
 base64 = "0.10"
 safemem = "0.3"
 smallvec = { version = "0.6", features = ["union"] }
-procinfo = { git = "https://github.com/tikv/procinfo-rs" }
 flate2 = { version = "1.0", features = ["zlib"], default-features = false }
 more-asserts = "0.1"
 hyper = "0.12"
@@ -143,6 +142,9 @@ arrow = "0.10.0"
 
 [target.'cfg(unix)'.dependencies]
 signal = "0.6"
+
+[target.'cfg(target_os = "linux")'.dependencies]
+procinfo = { git = "https://github.com/tikv/procinfo-rs" }
 
 [workspace]
 members = [

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -62,6 +62,7 @@ extern crate num;
 extern crate num_traits;
 #[macro_use]
 extern crate prometheus;
+#[cfg(target_os = "linux")]
 extern crate procinfo;
 extern crate prometheus_static_metric;
 extern crate protobuf;

